src/session.c

```diff
@@ -982,6 +982,15 @@ session_free(LIBSSH2_SESSION *session)
         LIBSSH2_FREE(session, session->remote.lang_prefs);
     }
 
+    /* Free startup_key_state.key_state_low variables */
```

Comment:
It seems like this should be cleaned up in kex.c so consumers (the session) doesn't need to know about implementation details. Without being able to reproduce this, it's hard to say what the best approach would be. I'd also suggest NULL'ing the freed memory for safety.

Reply:
You are probably right.
I haven't looked at this in some time, but as I recall, the problem was a memory leak that was found using valgrind.
We have an application that connects and disconnects all the time and noticed a slow memory leak that was tracked down to those objects not being freed.
I am not familiar with the kex architecture, but something should trigger freeing those when the session is destroyed. 
I can't even find any reference to the allocation of the startup_key_state object.